src/axom/quest/detail/shaping/WindingNumberSampler.hpp

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_QUEST_WINDING_NUMBER_SAMPLER__HPP_
+#define AXOM_QUEST_WINDING_NUMBER_SAMPLER__HPP_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/mint.hpp"
+#include "axom/spin.hpp"
+#include "axom/quest/InOutOctree.hpp"
+#include "axom/quest/detail/shaping/shaping_helpers.hpp"
+
+#include "axom/fmt.hpp"
+
+#include "mfem.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace shaping
+{
+
+using QFunctionCollection = mfem::NamedFieldsMap<mfem::QuadratureFunction>;
+using DenseTensorCollection = mfem::NamedFieldsMap<mfem::DenseTensor>;
+
+namespace detail
+{
+
+/*!
+ * \brief Check whether point \a pt is inside \a shape.
+ *
+ * \param shape The shape being checked for inside/outside.
+ * \param pt The point being checked against the shape.
+ *
+ * \return True if pt is inside shape; False otherwise.
+ */
+template <typename ShapeType, typename PointType>
+bool AXOM_HOST_DEVICE checkInside(const ShapeType& shape, const PointType& pt)
+{
+  bool inside = false;
+  double wn {};
+  for(int c = 0; c < shape.numEdges(); c++)
+  {
+    wn += axom::primal::winding_number(pt, shape[c]);
+  }
+  // A point inside the polygon should have non-zero winding number.
+  inside |= (std::round(wn) != 0);
+
+  return inside;
```

Comment:
Since the in/out is only determined once at the end, would this make sense?


```suggestion
  double wn {};
  for(int c = 0; c < shape.numEdges(); c++)
  {
    wn += axom::primal::winding_number(pt, shape[c]);
  }
  // A point inside the polygon should have non-zero winding number.
  return (std::round(wn) != 0);
```

Reply:
Fixed it. I had copied the code from an earlier version.